### DISCUSSION TITLE
How to upload a file to a strapi component?

### DISCUSSION THREAD

**Discussion by Moizsohail** [strapi/strapi#7949] [Legacy Topics]
I am trying to create an interface where users can make thier assignment submissions. In order to achieve that I have made use of components. However, I can't seem to figure out how to achieve this using components.

Please note that the following attempt allows the file to get successfully uploaded however it does not links with the appropriate submission.
My attempt to upload files
```
const submissions = [
      ...assignment.submissions.map((submission) => {
        delete submission._id;
        return submission;
      }),
      {
        comments: comments,
        links: links,
        status: "unread",
      },
    ];
    
    const entity = await strapi.services.assignment.update(
      { id: assignmentid },
      { submissions: submissions }
    );
    if (files) {
      // automatically uploads the files based on the entry and the model
      await strapi.entityService.uploadFiles(entity.submissions[1], files, {
        model: "assignment",
        // if you are using a plugin's model you will have to add the `source` key (source: 'users-permissions')
      });
    }
```
This is the structure of my submission component
![image](https://user-images.githubusercontent.com/32095770/93604571-f9f72100-f9de-11ea-9ca6-18044a282644.png)

**Comment by derrickmehaffy** [comment]
This closed issue may help you @Moizsohail https://github.com/strapi/strapi/issues/5961.

**Reply by Moizsohail** [reply]
Isn't there a way to link the file by modifying the 
```
 await strapi.entityService.uploadFiles(entity.submissions[1], files, {
        model: "assignment",
      });
```
The suggested issues recommends that we should send another post request to link them. Isn't there like a direct way

**Reply by derrickmehaffy** [reply]
You would modify the referenced model, not the file itself. (Passing the ID of the file to the other model's relational field)

**Reply by Moizsohail** [reply]
And how do I obtain the id of the file? Can you give an example code snippet on how to achieve this. I'm really confused

**Reply by derrickmehaffy** [reply]
The default upload route would probably be the best, what exactly are you trying to do?
